test/metabase/api/dashboard_test.clj

```diff
@@ -319,6 +321,66 @@
                                                    :series                     []}]})
                    (dashboard-response (mt/user-http-request :rasta :get 200 (format "dashboard/%d" dashboard-id)))))))))
 
+    (testing "a dashboard that has link cards on it"
+      (let [link-card-viz-setting (fn [model id]
+                                    {:virtual_card {:display "link"}
+                                     :link         {:entity {:id    id
+                                                             :model model}}})
+            link-card-info-from-resp (fn [resp]
+                                       (->> resp
+                                            :ordered_cards
+                                            (map #(get-in % [:visualization_settings :link :entity]))))
+            crowberto-pc-id         (db/select-one-field :id Collection :personal_owner_id (mt/user->id :crowberto))]
+        (t2.with-temp/with-temp
+          [Database      {db-id :id}        {:name        "Linked database"
+                                             :description "Linked database desc"}
+           Table         {table-id    :id}  {:db_id        db-id
+                                             :name        "Linked table"
+                                             :display_name "Linked table dname"
+                                             :description "Linked table desc"}
+           Card          {card-id :id}      {:name          "Linked card"
+                                             :description   "Linked card desc"
+                                             :display       "bar"
+                                             :collection_id crowberto-pc-id}
+           Card          {model-id :id}     {:dataset       true
+                                             :name          "Linked model"
+                                             :description   "Linked model desc"
+                                             :display       "table"
+                                             :collection_id crowberto-pc-id}
+           Dashboard     {dash-id :id}      {:name          "Linked Dashboard"
+                                             :description   "Linked Dashboard desc"
+                                             :collection_id crowberto-pc-id}
+           Dashboard     {dashboard-id :id} {:name        "Test Dashboard"}
+           DashboardCard _                  {:dashboard_id           dashboard-id
+                                             :visualization_settings (link-card-viz-setting "database" db-id)}
```

Comment:
Why is there no dashcard for links to collections?

Reply:
forgot about that, updated!